sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py

```diff
@@ -166,24 +166,39 @@ def __init__(self, **kwargs) -> None:
 
     @classmethod
     def _from_generated(
-        cls, response: "HttpResponse", deserialized_operation: "FullBackupOperation", response_headers: "Dict"
+        cls, response: HttpResponse, deserialized_operation: FullBackupOperation, response_headers: Dict
     ) -> "KeyVaultBackupResult":
         return cls(folder_url=deserialized_operation.azure_storage_blob_container_uri)
 
 
-class KeyVaultSetting(object):
+class KeyVaultSetting(Generic[T]):
     """A Key Vault setting.
 
     :ivar str name: The name of the account setting.
-    :ivar str value: The value of the pool setting.
-    :ivar SettingType type: The type specifier of the value.
+    :ivar value: The value of the pool setting. This is a boolean if `type` is SettingType.BOOLEAN, and a string
+        otherwise. If `value` is provided as a string when `type` is SettingType.BOOLEAN, the value will be converted
+        into a boolean (True if `value` is "True" (case-insensitive); False otherwise).
+    :vartype value: str or bool
+    :ivar type: The type specifier of the value.
+    :vartype type: SettingType or None
     """
 
-    def __init__(self, **kwargs) -> None:
-        self.name = kwargs.get("name")
-        self.value = kwargs.get("value")
-        self.type = kwargs.get("type")
+    def __init__(
+        self,
+        *,
+        name: str,
+        value: Union[str, bool],
```

Comment:
This should be typed according to the generic.

Reply:
I wonder if there's a way to link the type of T to a `Literal` value used for `type`.... Krista may know?

Reply:
I couldn't get the type checking to work with the generic type, unfortunately. Adding `bound` to the generic type definition didn't help either, so using a `Literal` might be the way to go